File: Java/app/src/main/java/com/androidnative/app/MainActivity.java
package com.androidnative.app;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;

import com.otpless.dto.OtplessRequest;
import com.otpless.dto.OtplessResponse;
import com.otpless.main.OtplessManager;
import com.otpless.main.OtplessView;
import com.otpless.utils.Utility;


public class MainActivity extends AppCompatActivity {
    OtplessView otplessView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        //******************************************************** */
        //This function will tell if WhatsApp is Installed or not.
        // If you are using only whatsapp authentication then you can hide visibility of Login button using this function,
        // if user doesn't have whatsapp installed on device.
        //******************************************************** */

        if (Utility.isWhatsAppInstalled(this)){
            Toast.makeText(this, "whatsapp is installed in device", Toast.LENGTH_SHORT).show();
        }else {
            Toast.makeText(this, "Whatsapp is not installed in device", Toast.LENGTH_SHORT).show();
        }


        // Initialise OtplessView
        otplessView = OtplessManager.getInstance().getOtplessView(this);
        OtplessRequest request = new OtplessRequest("appid")
                .addExtras("crossButtonHidden","true");
        otplessView.setCallback(request, this::onOtplessCallback);
        otplessView.showOtplessLoginPage(request, this::onOtplessCallback);
        otplessView.onNewIntent(getIntent());
    }

    private void onOtplessCallback(OtplessResponse response) {
        if (response.getErrorMessage() != null) {
    // todo error handing
        } else {
            final String token = response.getData().optString("token");
    // todo token verification with api
            Log.d("Otpless", "token: " + token);
            Toast.makeText(this, "Token : " + token, Toast.LENGTH_SHORT).show();
            Intent i = new Intent(getApplicationContext(), SecondActivity.class);
            i.putExtra("passing_token", token);
            startActivity(i);
        }
    }

    @Override
    protected void onNewIntent(Intent intent) {
        super.onNewIntent(intent);
        if (otplessView != null) {
            otplessView.onNewIntent(intent);
        }
    }
    @Override
    public void onBackPressed() {
        // make sure you call this code before super.onBackPressed();
        if (otplessView.onBackPressed()) return;
        super.onBackPressed();
    }
}

File: Kt/app/src/main/java/in/smiley/otpless/MainActivity.kt
package `in`.smiley.otpless

import android.content.ClipData
import android.content.ClipboardManager
import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.util.Log
import android.view.WindowInsetsController.APPEARANCE_LIGHT_STATUS_BARS
import android.widget.Button
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.google.gson.Gson
import com.google.gson.GsonBuilder
import com.otpless.dto.HeadlessChannelType
import com.otpless.dto.HeadlessRequest
import com.otpless.dto.HeadlessResponse
import com.otpless.dto.OtplessRequest
import com.otpless.dto.OtplessResponse
import com.otpless.main.OtplessManager
import com.otpless.main.OtplessView
import com.otpless.utils.Utility


class MainActivity : AppCompatActivity() {
    private lateinit var showPreBuildUIButton: Button
    private lateinit var whatsappButton: Button

    private lateinit var otplessView: OtplessView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        initializeOtpless()
        initializeViews()
        setOnClickListeners()
    }

    private fun initializeOtpless() {
        otplessView = OtplessManager.getInstance().getOtplessView(this)
    }

    private fun initializeViews() {
        if (Build.VERSION.SDK_INT > Build.VERSION_CODES.R) {
            window.decorView.windowInsetsController?.setSystemBarsAppearance(APPEARANCE_LIGHT_STATUS_BARS, APPEARANCE_LIGHT_STATUS_BARS)
        }

        showPreBuildUIButton = findViewById(R.id.btn_pre_built_ui)
        whatsappButton = findViewById(R.id.whatsapp)

    }

    private fun setOnClickListeners() {
        showPreBuildUIButton.setOnClickListener {
            showPreBuiltUI()
        }

        whatsappButton.setOnClickListener {
            startActivity(Intent(this, HeadlessActivity::class.java))
        }
    }

    private fun showPreBuiltUI() {
        val request = OtplessRequest("appid")
        otplessView.setCallback(request, this::onOtplessCallback)
        otplessView.showOtplessLoginPage(request, this::onOtplessCallback)
        otplessView.verifyIntent(intent)
    }


    private fun onOtplessCallback(response: OtplessResponse) {

        val gson: Gson = GsonBuilder().setPrettyPrinting().create()
        val jsonResponse = if (response.errorMessage != null) {
            gson.toJson(response.errorMessage)
        } else {
            gson.toJson(response.data)
        }

        val clipboard: ClipboardManager = getSystemService(CLIPBOARD_SERVICE) as ClipboardManager
        val clip = ClipData.newPlainText("Otpless Response", jsonResponse)
        clipboard.setPrimaryClip(clip)
        Toast.makeText(this, "Copied Successfully", Toast.LENGTH_SHORT).show()

        val intent = Intent(this, ResponseActivity::class.java)
        intent.putExtra("RESPONSE", jsonResponse)
        startActivity(intent)
        Log.d("OtplessCallback", "Response: $jsonResponse")
    }

    override fun onNewIntent(intent: Intent?) {
        super.onNewIntent(intent)
        otplessView.verifyIntent(intent)
    }

    override fun onBackPressed() {
        if (otplessView.onBackPressed()) return
        super.onBackPressed()
    }
}